fast-paginate 组件报错

```
2024-12-11 14:20:46.106 [ERROR] [AppExceptionHandler::handle] TypeError: Hyperf\Database\Model\Model::newCollection(): Argument #1 ($models) must be of type array, Hyperf\Database\Model\Collection given, called in /WWW/php-service/vendor/friendsofhyperf/fast-paginate/src/FastPaginate.php on line 30 and defined in /WWW/php-service/vendor/hyperf/database/src/Model/Model.php:858
Stack trace:
#0 /WWW/php-service/vendor/friendsofhyperf/fast-paginate/src/FastPaginate.php(30): Hyperf\Database\Model\Model->newCollection(Object(Hyperf\Database\Model\Collection))
#1 [internal function]: Hyperf\Database\Model\Builder->FriendsOfHyperf\FastPaginate\{closure}(Object(Hyperf\Database\Model\Collection), Object(Hyperf\Paginator\LengthAwarePaginator))
#2 /WWW/php-service/vendor/friendsofhyperf/fast-paginate/src/FastPaginate.php(172): Closure->call(Object(Hyperf\Database\Model\Builder), Object(Hyperf\Database\Model\Collection), Object(Hyperf\Paginator\LengthAwarePaginator))
#3 [internal function]: Hyperf\Database\Model\Builder->FriendsOfHyperf\FastPaginate\{closure}(5)
#4 /WWW/php-service/vendor/hyperf/database/src/Model/Builder.php(152): call_user_func_array(Object(Closure), Array)
#5 /WWW/php-service/runtime/container/proxy/App_Controller_MysqlController.proxy.php(206): Hyperf\Database\Model\Builder->__call('fastPaginate', Array)
#6 /WWW/php-service/runtime/container/proxy/Hyperf_HttpServer_CoreMiddleware.proxy.php(131): App\Controller\MysqlController->page3()
#7 /WWW/php-service/runtime/container/proxy/Hyperf_HttpServer_CoreMiddleware.proxy.php(87): Hyperf\HttpServer\CoreMiddleware->handleFound(Object(Hyperf\HttpServer\Router\Dispatched), Object(Hyperf\HttpMessage\Server\Request))
#8 /WWW/php-service/vendor/hyperf/dispatcher/src/AbstractRequestHandler.php(45): Hyperf\HttpServer\CoreMiddleware->process(Object(Hyperf\HttpMessage\Server\Request), Object(Hyperf\Dispatcher\HttpRequestHandler))
#9 /WWW/php-service/vendor/hyperf/dispatcher/src/HttpRequestHandler.php(27): Hyperf\Dispatcher\AbstractRequestHandler->handleRequest(Object(Hyperf\HttpMessage\Server\Request))
#10 /WWW/php-service/app/Middleware/CorsMiddleware.php(37): Hyperf\Dispatcher\HttpRequestHandler->handle(Object(Hyperf\HttpMessage\Server\Request))
#11 /WWW/php-service/vendor/hyperf/dispatcher/src/AbstractRequestHandler.php(45): App\Middleware\CorsMiddleware->process(Object(Hyperf\HttpMessage\Server\Request), Object(Hyperf\Dispatcher\HttpRequestHandler))
#12 /WWW/php-service/vendor/hyperf/dispatcher/src/HttpRequestHandler.php(27): Hyperf\Dispatcher\AbstractRequestHandler->handleRequest(Object(Hyperf\HttpMessage\Server\Request))
#13 /WWW/php-service/vendor/hyperf/tracer/src/Middleware/TraceMiddleware.php(56): Hyperf\Dispatcher\HttpRequestHandler->handle(Object(Hyperf\HttpMessage\Server\Request))
#14 /WWW/php-service/vendor/hyperf/dispatcher/src/AbstractRequestHandler.php(45): Hyperf\Tracer\Middleware\TraceMiddleware->process(Object(Hyperf\HttpMessage\Server\Request), Object(Hyperf\Dispatcher\HttpRequestHandler))
#15 /WWW/php-service/vendor/hyperf/dispatcher/src/HttpRequestHandler.php(27): Hyperf\Dispatcher\AbstractRequestHandler->handleRequest(Object(Hyperf\HttpMessage\Server\Request))
#16 /WWW/php-service/runtime/container/proxy/Hyperf_Dispatcher_HttpDispatcher.proxy.php(38): Hyperf\Dispatcher\HttpRequestHandler->handle(Object(Hyperf\HttpMessage\Server\Request))
#17 /WWW/php-service/vendor/hyperf/di/src/Aop/ProceedingJoinPoint.php(57): Hyperf\Dispatcher\HttpDispatcher->Hyperf\Dispatcher\{closure}(Object(Hyperf\HttpMessage\Server\Request), Array, Object(Hyperf\HttpServer\CoreMiddleware))
#18 /WWW/php-service/vendor/hyperf/di/src/Aop/ProxyTrait.php(87): Hyperf\Di\Aop\ProceedingJoinPoint->processOriginalMethod()
#19 /WWW/php-service/vendor/hyperf/pipeline/src/Pipeline.php(98): Hyperf\Dispatcher\HttpDispatcher::Hyperf\Di\Aop\{closure}(Object(Hyperf\Di\Aop\ProceedingJoinPoint))
#20 /WWW/php-service/vendor/hyperf/di/src/Aop/ProceedingJoinPoint.php(46): Hyperf\Pipeline\Pipeline::Hyperf\Pipeline\{closure}(Object(Hyperf\Di\Aop\ProceedingJoinPoint))
#21 /WWW/php-service/vendor/friendsofhyperf/telescope/src/Aspect/RequestDispatcherAspect.php(36): Hyperf\Di\Aop\ProceedingJoinPoint->process()
#22 /WWW/php-service/vendor/hyperf/di/src/Aop/Pipeline.php(31): FriendsOfHyperf\Telescope\Aspect\RequestDispatcherAspect->process(Object(Hyperf\Di\Aop\ProceedingJoinPoint))
#23 /WWW/php-service/vendor/hyperf/pipeline/src/Pipeline.php(81): Hyperf\Di\Aop\Pipeline->Hyperf\Di\Aop\{closure}(Object(Hyperf\Di\Aop\ProceedingJoinPoint))
#24 /WWW/php-service/vendor/hyperf/di/src/Aop/ProxyTrait.php(88): Hyperf\Pipeline\Pipeline->then(Object(Closure))
#25 /WWW/php-service/vendor/hyperf/di/src/Aop/ProxyTrait.php(31): Hyperf\Dispatcher\HttpDispatcher::handleAround(Object(Hyperf\Di\Aop\ProceedingJoinPoint))
#26 /WWW/php-service/runtime/container/proxy/Hyperf_Dispatcher_HttpDispatcher.proxy.php(39): Hyperf\Dispatcher\HttpDispatcher::__proxyCall('Hyperf\\Dispatch...', 'dispatch', Array, Object(Closure))
#27 /WWW/php-service/vendor/hyperf/http-server/src/Server.php(112): Hyperf\Dispatcher\HttpDispatcher->dispatch(Object(Hyperf\HttpMessage\Server\Request), Array, Object(Hyperf\HttpServer\CoreMiddleware))
#28 [internal function]: Hyperf\HttpServer\Server->onRequest(Object(Swoole\Http\Request), Object(Swoole\Http\Response))
#29 {main}

```


<img width="1295" alt="image" src="https://github.com/user-attachments/assets/d000a96a-fbe7-440d-9f62-2bcbecef46a1">

$items 需要传数组

```
Hyperf\Database\Model\Model::newCollection(): Argument #1 ($models) must be of type array, Hyperf\Database\Model\Collection given
```

--- a/src/fast-paginate/src/FastPaginate.php
+++ b/src/fast-paginate/src/FastPaginate.php
@@ -167,7 +167,7 @@ class FastPaginate
             // The $paginator is full of records that are primary keys only. Here,
             // we create a new paginator with all of the *stats* from the index-
             // only paginator, but the *items* from the outer query.
-            $items = $this->simplePaginate($perPage, $columns, $pageName, 1)->getCollection(); /* @phpstan-ignore-line */
+            $items = $this->simplePaginate($perPage, $columns, $pageName, 1)->items();
 
             return Closure::fromCallable($paginatorOutput)->call($this, $items, $paginator);
         };

